inputremapper/injection/consumers/mapping_parser.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# input-remapper - GUI for device specific keyboard mappings
+# Copyright (C) 2022 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of input-remapper.
+#
+# input-remapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# input-remapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with input-remapper.  If not, see <https://www.gnu.org/licenses/>.
+"""functions to assemble the mapping handlers"""
+
+from typing import Dict, List, Type
+
+from evdev.ecodes import EV_KEY, EV_ABS, EV_REL
+
+from inputremapper.logger import logger
+from inputremapper.key import Key
+from inputremapper.injection.consumers.mapping_handler import (
+    MappingHandler,
+    CombinationHandler,
+    ContextProtocol,
+    HierarchyHandler,
+    AbsToBtnHandler,
+    RelToBtnHandler,
+)
+from inputremapper.mapping import Mapping
+
+MappingHandlers = Dict[Key, List[MappingHandler]]
+
+mapping_handler_classes: Dict[str, Type[MappingHandler]] = {
+    # all available mapping_handlers
+    "combination": CombinationHandler,
+}
+
+
+def parse_mapping(mapping: Mapping, context: ContextProtocol) -> MappingHandlers:
```

Comment:
I think the `Mapping` class should be renamed to `Preset`

Reply:
I was thinking something like this as well. A preset is a collection of mappings, so renaming this would fit well.